### PR TITLE
Develop/frontend

### PR DESCRIPTION
**Pull request by nakajima-john-shotaro**
## Pull Request Type

- [ ] :bug: `fix` : (bug fix for the user, not a fix to a build script)
- [x] :adhesive_bandage: `change` : (simple fix for a non-critical issue)
- [x] :sparkles: `feat` : (new feature for the user, not a new feature for build script)
- [ ] :memo: `docs` : (changes to the documentation)
- [ ] :recycle: `refactor` : (refactoring production code, eg. renaming a variable)
- [ ] :pencil2: `style` : (formatting, missing semi colons, etc; no production code change)
- [ ] :rotating_light: `test` : (adding missing tests, refactoring tests; no production code change)
- [ ] :package: `chore` : (updating grunt tasks etc; no production code change)
- [ ] :rewind: `reverts` : (canceling commits or pull requests)

## Description

Add new alerts and make minor changes to display.

## Pre-Review Checklist for the Pull Request Author

- [x] Code follows coding-guidelines
- [x] Assign Pull Request to reviewer

## Checklist for the Pull Request Reviewer

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows coding-guidelines
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] Pull Request title describes the changes

## Post-Review Checklist for the Pull Request Author

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write release notes

### CODE REVIEW

File: aicon/frontend/static/js/functions.js [hunk 1]
```diff
@@ -753,18 +778,43 @@ function notify_alert(alart_title, alart_text, reload=true) {
     });
 }
 
+function test_device_status(device_status) {
+    let alert_text = "";
+    let check_list = [];
+    let icon = 'warning';
+    let binary_device_status =device_status.toString(2);
```
Comment (urasakikeisuke) [inline aicon/frontend/static/js/functions.js]:
```suggestion
    let binary_device_status = device_status.toString(2);
```

File: aicon/frontend/static/js/functions.js [hunk 2]
```diff
@@ -534,17 +550,26 @@ function check() {
...
         timeout: 10000,
         async: true, // 同期通信  false:同期  true:非同期
         contentType: "application/json; charset=utf-8",
     })
         .done(function () {
             communication_status = true;
             check();
+            if (r_data['pre_diagnostics'] !== 0){
+                let pre_check_list = test_device_status(r_data['pre_diagnostics']);
+                let notice_sting = "下記の項目に注意してください。\n" + pre_check_list[0];
```
Comment (urasakikeisuke) [inline aicon/frontend/static/js/functions.js]: ここ直ってないような気がします。

File: aicon/frontend/static/js/functions.js [hunk 3]
```diff
@@ -534,17 +550,26 @@ function check() {
 // サーバーIPアドレスとの通信が可能かを確認
 var communication_status = false;
 $('#communication_partner').focusout(function(){
+    send_data = {
+        version: icon_varsion
+    }
```
Comment (urasakikeisuke) [inline aicon/frontend/static/js/functions.js]:
```suggestion
    };
```

File: aicon/frontend/static/js/functions.js [hunk 4]
```diff
@@ -9,28 +10,43 @@ $(window).on('load resize', function () {
...
             communication_status = true;
+            if (!$.cookie('first_login')) {
+                if (r_data['pre_diagnostics'] !== 0){
+                    let pre_check_list = test_device_status(r_data['pre_diagnostics']);
+                    let notice_sting = "Something went wrong.\n" + pre_check_list[0];
+                    notify_alert("Oops!", notice_sting, false, pre_check_list[1]);
+                }
+                $.cookie('first_login', 'no');
+            }
         })
```
Comment (urasakikeisuke) [inline aicon/frontend/static/js/functions.js:44]:
```suggestion
        });
```

File: aicon/frontend/static/js/functions.js [hunk 5]
```diff
@@ -9,28 +10,43 @@ $(window).on('load resize', function () {
...
     change_advanced_param(model_button_id, param_button_id);
     $('#textarea').focus();
     $('#communication_partner').val($.cookie('url'));
     if ($('#communication_partner').val() == ''){
         $('#communication_partner').val('localhost');
     }
     $('html,body').animate({ scrollTop: 0 }, '1');
+    send_data = {
+        version: icon_varsion
+    }
```
Comment (urasakikeisuke) [inline aicon/frontend/static/js/functions.js]:
```suggestion
    };
```

File: aicon/frontend/static/js/functions.js [hunk 6]
```diff
@@ -1,3 +1,4 @@
+var icon_varsion = '1.2.1';
```
Comment (urasakikeisuke) [inline aicon/frontend/static/js/functions.js]: 定数は大文字でお願いします

**Review by urasakikeisuke — CHANGES_REQUESTED** [review]
よろしくお願いします

File: aicon/frontend/static/js/functions.js (hunk 1 above)
Comment (nakajima-john-shotaro) [reply aicon/frontend/static/js/functions.js]: 早急に対応しようと思います。
いつも丁寧な確認ありがとうございます
HamsterCuteGIF
。

File: aicon/frontend/static/js/functions.js (hunk 4 above)
Comment (nakajima-john-shotaro) [reply aicon/frontend/static/js/functions.js:44]: ここに；は厳禁です。👺

**Comment by nakajima-john-shotaro** [comment]
変更しました。確認の方よろしくお願いします。

File: aicon/frontend/static/js/functions.js (hunk 4 above)
Comment (urasakikeisuke) [reply aicon/frontend/static/js/functions.js:44]: すみませんでした。

**Review by urasakikeisuke — APPROVED** [review]
FriendsRossGIF

**Pull request merged**